spinn_front_end_common/interface/provenance/fec_timer.py

```diff
@@ -65,51 +69,85 @@ class FecTimer(object):
         # Name of algorithm what is being timed
         "_algorithm",
 
-        # Name of category what is being timed
-        "_category",
+        # Type of work being done
+        "_work"
         ]
 
+    # Category Constants
+    WAITING = "Waiting"
+    SETTING_UP = "In Setup"
+    RUN_OTHER = "In run other"
+    GET_MACHINE = "Turning on Machine"
+    LOADING = "Loading Stage"
+    DATA_GENERATION = "data_generation"
+    MAPPING = "Mapping Stage"
+    RUN_LOOP = "Running Stage"
+    BUFFER = "BufferExtraction"
+    RESETTING = "Resetting"
+    SHUTTING_DOWN = "Shutting down"
+    ERROR = "Error"
+
+    CATEGORIES = [WAITING, SETTING_UP, RUN_OTHER, GET_MACHINE, MAPPING,
+                  LOADING, RUN_LOOP, BUFFER, RESETTING, SHUTTING_DOWN, ERROR]
+
+    # Work Type Constants
+    OTHER = "Other"
+    # LOADING
+    BITFIELD = "BitField work"
+    # Only for on Machine Compression
+    COMPRESSING = "Compressing"
+    CONTROL = "Control"
+    SYNAPSE = "Expanding Synapse"
+    RUNNING = "Running"
+    EXTRACTING = "Extracting"
+    REPORT = "Reporting"
+
+    WORK_TYPES = [
+        OTHER, GET_MACHINE, LOADING, BITFIELD, CONTROL, SYNAPSE, RUNNING,
+        EXTRACTING, REPORT]
```

Comment:
See above - might be better as an Enum

Reply:
yep

Reply:
Created TimerCaterory and TimerWork enums